code/script.js

```diff
@@ -255,71 +294,125 @@ const checkQuestion = () => {
+     alert(
         `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
       )
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } 
+  else if (category === 'other' || category === 'gender') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person is a ${value}! Keep all the ${value}s`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person is not a ${value}! Remove all the ${value}s`
+      )     
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+ 
+    // for hair and eyes :
+    if (category === 'hair' || category === 'eyes') {
+      if (keep) {
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } else {
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+// for accessories and other
+    } else {
+      if (keep) {
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+      } else {
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      }
+      }
+      
+     // Invoke a function to redraw the board with the remaining people.
+     generateBoard() 
 }
+start()
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+  const playerGuess = confirm(`Do you really want to guess on ${personToConfirm}?`)
   // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+   // If the player wants to guess, invoke the checkMyGuess function.
+  if (playerGuess) {
+    checkMyGuess(personToConfirm)
+  }
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+// 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
```

Comment:
I think this part is maybe a little bit confusing. Maybe you could have just cleaned up the "empty" comments and written new ones that makes more scens for your specific code.